indolaw-parser/parser_utils.py

```diff
@@ -624,6 +678,49 @@ def g(node: Union[ComplexNode, PrimitiveNode]):
 
     g(undang_undang_node)
 
+    ketentuan_umum: Dict[str, Any] = {}
+
+    def parse_ketentuan_umum(node: Union[ComplexNode, PrimitiveNode]):
+        if not node.children:
+            return
+
+        if len(node.children) <= 2:
+            for child in node.children:
+                if isinstance(child, PrimitiveNode) and child.type == Structure.PLAINTEXT:
+                    definition_text = child.text.split(r'adalah')
+
+                    title = definition_text[0].strip()
+                    definition = definition_text[1].strip()
+
+                    ketentuan_umum[title] = definition
+        else:
+            title = None
+            definition_list = []
+
+            for child in node.children:
+                if child.type == Structure.PLAINTEXT:
+                    title = child.text.split(r'adalah')[0].strip()
+                elif child.type == Structure.LIST:
+                    for list_item in child.children:
+                        definition = " ".join([node.text for node in list_item.children])
+                        definition_list.append(definition)
```

Comment:
how reliable is the heuristic here? does it work 100% of the time? if not, how often? and if the heuristic fails, how do we work around it?

Reply:
This should be reliable for most cases. Haven't found any differing edge cases yet. (Either is "x adalah y", or "x adalah:" followed by a list of definitions)